Remove the use of tito

ehelms opened the pull request:
With the change to using Koji CLI directly for build and release,
the tito configuration is unnecessary to continue managing.

<!--
If your package needs to be released within one or more release streams, and/or distributions, please open PRs to each of those branches respectively. The easiest way to do this is to make the initial commit for the mainline branch (e.g. rpm/develop or deb/develop) and then cherry pick the commit hash onto each subsequent branch.

Supported Versions:

 * Nightly
 * 2.0
 * 1.24
 * 1.23
 
RPM Example:

    git checkout -b rpm/develop-foreman-tasks-1.0.1 rpm/develop

    # Make changes to update package

    git commit -a -m 'Release foreman-tasks-1.0.1'
    COMMIT=`git rev-parse HEAD`

    git checkout -b rpm/1.20-foreman-tasks-1.0.1-1.20 rpm/1.20
    git cherry-pick -x $COMMIT

DEB Example:

    git checkout -b deb/develop-foreman-tasks-1.0.1 deb/develop

    # Make changes to update package

    git commit -a -m 'Release foreman-tasks-1.0.1'
    COMMIT=`git rev-parse HEAD`

    git checkout -b deb/1.20-foreman-tasks-1.0.1-1.20 deb/1.20
    git cherry-pick -x $COMMIT

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.
-->

evgeni commented on README.md:
```diff
@@ -23,9 +23,7 @@ If you're submitting a patch which adds/updates npm modules (nodejs- packages) y
 To build locally or release RPMs from this repo, you also require:
 
 * [obal](https://github.com/theforeman/obal) 0.0.2 or higher
```
You need an obal release with the un-tito-fication in, which doesn't exist yet ;)

ekohl commented on package_manifest.yaml:
```diff
@@ -248,14 +240,14 @@ foreman_core_packages:
-        - koji-foreman-plugins
+      koji_tags:
+        - "{{ all_tags['foreman-el8'] }}"
+        - "{{ all_tags['foreman-plugins-el8'] }}"
     rubygem-net_http_unix: {}
     rubygem-netrc:
-      releasers:
-        - koji-foreman
-        - koji-foreman-plugins
+      koji_tags:
+        - "{{ all_tags['foreman-el8'] }}"
+        - "{{ all_tags['foreman-plugins-el8'] }}"
```
Why do we release these to both foreman and foreman-plugins?

ehelms replied:
Something something SCl...goes looking.... "Also build rubygem-net-ssh for non-SCL " so I think this means we can stop this.

ekohl commented on list_koji_tags.py:
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+
+from ruamel.yaml import YAML
```
If you're not writing YAML, I'd use vanilla yaml
```suggestion
import yaml
```

ekohl commented on list_koji_tags.py:
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
```
I'd not use a file extension so you're free to change the implementation language.

ekohl commented on add_pypi_package.sh:
```diff
@@ -105,13 +94,13 @@ add_pypi_to_comps() {
+	if [[ $KOJI_TAG == "foreman-nightly-nonscl-rhel7" ]] ; then
 		local section="foreman_nonscl_packages"
-	elif [[ $TITO_TAG == "foreman-plugins-nightly-nonscl-rhel7" ]] ; then
+	elif [[ $KOJI_TAG == "foreman-plugins-nightly-nonscl-rhel7" ]] ; then
 		local section="plugin_nonscl_packages"
-	elif [[ $TITO_TAG == "katello-nightly-rhel7" ]] ; then
+	elif [[ $KOJI_TAG == "katello-nightly-rhel7" ]] ; then
 		local section="katello_packages"
-	elif [[ $TITO_TAG == "katello-pulpcore-nightly-el7" ]] ; then
+	elif [[ $KOJI_TAG == "katello-pulpcore-nightly-el7" ]] ; then
 		local section="pulpcore_packages"
```
This should be updated to EL8. Alternatively, we can not talk about a koji tag but require the user to specify the section in the manifest.

ehelms replied:
I thought about this as well and figured it as a future update to persue on top of this change.

ekohl reviewed (commented):
So how do we deal with package whitelisting in Koji after this?

ehelms commented:
> So how do we deal with package whitelisting in Koji after this?

Checking the status of? Actually doing it? What do you mean specifically here?

ehelms commented:
> Do we continue using the whitelist? If so, how do we ensure new packages are whitelisted?
> 
> I enjoyed using `rel-eng/compare-to-tags` to see if the whitelists were correct. If we continue using whitelists, I'd like to see a replacement.

I am not sure *which* whitelist you mean here: koji or tito. Koji enforces whitelisting as packages must be added to a tag. We are not managing that in anyway today to my knowledge. Tito whitelisting was a filtering mechanism for identifying which packages should be built against what tags -- this is handled by `koji_tags` property with Obal.

ekohl commented:
Right, but the script `rel-eng/compare-to-tags` gave an overview of which Koji packages were whitelisted but shouldn't according to the tito config and vice versa: which weren't whitelisted but should. This helped to clean things up.

So what I'm essentially asking for (since we'll keep the koji whitelist) is a way to gather what should be in each koji tag and then diff it against what's actually in the tag. Does that make sense?

ehelms commented:
> Right, but the script `rel-eng/compare-to-tags` gave an overview of which Koji packages were whitelisted but shouldn't according to the tito config and vice versa: which weren't whitelisted but should. This helped to clean things up.
> 
> So what I'm essentially asking for (since we'll keep the koji whitelist) is a way to gather what should be in each koji tag and then diff it against what's actually in the tag. Does that make sense?

Last time this was requested the answer I put together was -- https://github.com/theforeman/obal/pull/243

The other idea I had been thinking about was moving tag management into this repository from tool_belt and using that to declare and manage the Koji whitelist tightly -- https://github.com/ktdreyer/koji-ansible#koji_tag

ehelms commented:
> Last time this was requested the answer I put together was -- [theforeman/obal#243](https://github.com/theforeman/obal/pull/243)

Using this functionality locally, as the solution you'd get the following output right now (which identifies a few actions):

```
obal-source verify-koji-tag all

PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************************
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/all.yaml for localhost

TASK [verify_koji_tag : Find all tags] ***************************************************************************************************************************************************************************************************************************************************
ok: [localhost]
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for foreman-client-nightly-el8] **************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - jsoncpp
  - qpid-proton

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-client-nightly-el9] ***************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-client-nightly-el9 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-client-nightly-rhel7] *************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-client-nightly-rhel7 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-nightly-el8] **********************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-nightly-el8 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for foreman-plugins-nightly-el8] *************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - yggdrasil

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for katello-nightly-el8] *********************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - rubygem-pulp_2to3_migration_client
  - rubygem-redhat_access
  - rubygem-redhat_access_lib

TASK [verify_koji_tag : Packages missing in koji for katello-nightly-el8] ****************************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - rubygem-qpid_proton
```

ekohl commented:
I know you wanted to get this in before 3.4 branched, but I haven't gotten around to it. Can we get RC1 out and then see how this goes in nightly? I'd be happy to cherry pick it before RC2 if it works well.

ehelms commented:
> I know you wanted to get this in before 3.4 branched, but I haven't gotten around to it. Can we get RC1 out and then see how this goes in nightly? I'd be happy to cherry pick it before RC2 if it works well.

Sure? Everything in here related to tito is just noise in the repository at this point as tito is not used in any of the build paths. The other changes make things cleaner / nicer or more accurate with the current source of truth.

ehelms commented:
The corresponding Obal functionality for tag verification has been merged.

ehelms commented:
Rebased and cleaned this up

ekohl reviewed (changes requested):
I have since introduced https://github.com/theforeman/foreman-packaging/blob/rpm/develop/remove_package. Could you also modify that?

ehelms commented on README.md:
```diff
@@ -46,27 +44,17 @@ Configuration for mock is supplied in mock/ and can be used to build any of
 
     obal scratch PACKAGE
 
-Using a local git checkout, change `source_dir` as appropriate:
-
-* Core packages: `obal scratch foreman -e "{'releasers':['koji-foreman-jenkins'], 'build_package_tito_releaser_args': ['--arg source_dir=~/foreman']}"`
-* Plugins: `obal scratch rubygem-foreman_bootdisk -e "{'releasers': ['koji-foreman-plugins-jenkins']}, 'build_package_tito_releaser_args': ['--arg source_dir=~/foreman_bootdisk']}"`
-* Katello: `obal scratch rubygem-katello -e "{'releasers': ['koji-katello-jenkins'], 'build_package_tito_releaser_args': ['--arg source_dir=~/katello']}"`
```
That is correct, that is functionality that would need to be added to Obal.

ehelms commented on add_gem_package.sh:
```diff
@@ -1,9 +1,9 @@
 
 usage() {
-	echo "Usage: $0 GEM_NAME [TEMPLATE [TITO_TAG [PACKAGE_SUBDIR]]]"
+	echo "Usage: $0 GEM_NAME [TEMPLATE [KOJI_TAG [PACKAGE_SUBDIR]]]"
```
Given the `package_manifest` sections can be changed or split out into smaller ones that make sense to update or group together and I am hesitant to tie the updates to that just yet. I'd rather build up more first class support for identifying and calculating it.

ehelms commented:
Rebased on latest breaking changes, updates to `rel-eng` will keep breaking this PR so I'd like to merge it.

ehelms commented:
Rebased again - any final comments before I merge?
